src/plot_utils.py

```diff
@@ -100,20 +97,47 @@ def heatmap(bdb, bql, selectors=None, **kwargs):
     -------
     clustermap: seaborn.clustermap
     """
-    df = bqlu.cursor_to_df(bdb.execute(bql))
+    assert bql is not None or df is not None
+    if bql is not None:
+        df = bqlu.cursor_to_df(bdb.execute(bql))
+    df.fillna(0, inplace=True)
+    return zmatrix(df, **kwargs)
+
+def selected_heatmaps(bdb, selectors, bql=None, df=None, **kwargs):
+    """Plot heatmaps of pairwise matrix, broken up according to selectors.
+
+    Parameters
+    ----------
+    bdb : bayeslite.BayesDB
+        Active BayesDB instance.
+    selectors : [lambda name --> bool]
+        Rather than plot the full NxN matrix all together, make separate plots
+        for each combination of these selectors, plotting them in sequence.
+        If selectors are specified, yields clustermaps, which caller is
+        responsible for showing or saving, and then closing.
+    bql : str
+        The BQL to run and plot. Must be a PAIRWISE BQL query if specified.
+    df : pandas.DataFrame(columns=['generator_id', 'name0', 'name1', 'value'])          If bql is not specified, then take data from here instead.
+    **kwargs : dict
+        Passed to zmatrix: vmin, vmax, row_ordering, col_ordering
+
+    Yields
+    ------
+    The triple (clustermap, selector1, selector2).  It is recommended that
```

Comment:
Why use the identities of the functions themselves, rather than their indices in the selector list?  I think that would add about four tokens to the implementation.

Alternately, could accept a list of pairs (id, selector) and yield the ids, or perhaps a dict id : selector (in the latter case, would want to traverse it in sorted order by id, for determinism).